src/txn/txn.c

```diff
@@ -944,6 +942,13 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 		fileid = op->btree->id;
 		switch (op->type) {
 		case WT_TXN_OP_NONE:
+			/*
+			 * Increment the resolved update count for empty
+			 * operations to avoid thinking we've lost a prepared
+			 * update as we've got less resolved updates than
+			 * the txn mod count.
+			 */
```

Comment:
What is the reason to have empty operations? A comment will help.
Also, I see now that since you increment `resolved_update_count` here, you do not want to zero out `resolved_update_count` inside `__txn_resolve_prepared_update()`. Perhaps this could have been a different local variable and the value obtained from `__txn_resolve_prepared_update()` gets added into it. I expected `__wt_txn_resolve_prepared_op()` to output number of prepared updates, not increment the value by it. Maybe it just needs improving comments.